src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -1254,58 +1254,52 @@ final int queueSize() {
          * @throws RejectedExecutionException if array could not be resized
          */
         final void push(ForkJoinTask<?> task, ForkJoinPool pool, boolean internal) {
-            int s = top, b = base, m, cap, room; ForkJoinTask<?>[] a, na;
-            if ((a = array) != null && (cap = a.length) > 0) { // else disabled
-                int k = (m = cap - 1) & s;
-                if ((room = m - (s - b)) >= 0) {
-                    top = s + 1;
-                    long pos = slotOffset(k);
-                    if (!internal)
-                        U.putReference(a, pos, task);       // inside lock
-                    else
-                        U.getAndSetReference(a, pos, task); // fully fenced
-                    if (room == 0 && (na = growArray(a, cap, s)) != null)
-                        k = ((a = na).length - 1) & s;      // resize
-                }
+            int s = top++;                      // back out on failure
+            ForkJoinTask<?>[] a = array;
+            int size = s - base + 1, m;
+            if (((a != null && a.length > size) || (a = growArray(a, s)) != null) &&
```

Comment:
Instruction ordering reason to assign the size ahead of the `if`?  🤔 

```
            int m;
            if (((a != null && a.length > (s - base + 1)) || (a = growArray(a, s)) != null) &&
```

Reply:
(Now reworked a little...) Well, mainly trying to simply the conditional while avoiding possibly-uninitialized errors